boards/arm/stm32f412g_disco/arduino_r3_connector.dtsi

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2020 Kim Bøndergaard <kim.boendergaard@escoglobal.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	arduino_header: connector {
+		compatible = "arduino-header-r3";
+		#gpio-cells = <2>;
+		gpio-map-mask = <0xffffffff 0xffffffc0>;
+		gpio-map-pass-thru = <0 0x3f>;
+		gpio-map = <0 0 &gpioa 1 0>,	/* A0 */
+			   <1 0 &gpioc 1 0>,	/* A1 */
+			   <2 0 &gpioc 3 0>,	/* A2 */
+			   <3 0 &gpioc 4 0>,	/* A3 */
+			   <4 0 &gpioc 5 0>,	/* A4 */
+			   <5 0 &gpiob 0 0>,	/* A5 */
+			   <6 0 &gpiog 9 0>,	/* D0 */
+			   <7 0 &gpiog 14 0>,	/* D1 */
+			   <8 0 &gpiog 13 0>,	/* D2 */
+			   <9 0 &gpiof 4 0>,	/* D3 */
+			   <10 0 &gpiog 12 0>,	/* D4 */
+			   <11 0 &gpiof 10 0>,	/* D5 */
+			   <12 0 &gpiof 3 0>,	/* D6 */
+			   <13 0 &gpiog 11 0>,	/* D7 */
+			   <14 0 &gpiog 10 0>,	/* D8 */
+			   <15 0 &gpiob 8 0>,	/* D9 */
+			   <16 0 &gpioa 15 0>,	/* D10 */
+			   <17 0 &gpioa 7 0>,	/* D11 */
+			   <18 0 &gpioa 6 0>,	/* D12 */
+			   <19 0 &gpioa 5 0>,	/* D13 */
+			   <20 0 &gpiob 9 0>,	/* D14 */
+			   <21 0 &gpiob 10 0>;	/* D15 */
+	};
+};
+
+arduino_i2c: &i2c2 {
```

Comment:
Generally, only alias is in arduino_r3_connector.dtsi, the i2c, spi and usart should be defined in stm32f412g_disco.dts.
Also, missing 	`status = "okay";` for i2c, spi and usart.

supported of arduino_gpio, arduino_i2c and arduino_spi, can be added to nucleo_l476rg.yaml
(see nucleo_l476rg as example)

Reply:
I deliberately left out the Okay part. There might be situations where you don't want to activate e.g. the SPI pins but use an alternate function.
Regarding where to place i2c, spi and usart: 
This MCU has several pins you can use for e.g. spi1. Aliasing arduino_spi to spi1 is only valid with exactly this pin configuration even though the aliasing is probably independent of the pin settings.
I'll have a look at nucleo_l476rg

Reply:
> There might be situations where you don't want to activate e.g. the SPI pins but use an alternate function.

What we intend to provide here is a default board configuration.
cf https://docs.zephyrproject.org/latest/guides/porting/board_porting.html#general-recommendations

Reply:
Yeah I see that. Will fix it